app/code/core/Mage/Adminhtml/Block/Widget/Grid.php

```diff
@@ -313,9 +313,6 @@ public function addColumn($columnId, $column)
                 ->setData($column)
                 ->setGrid($this);
         }
-        /*elseif ($column instanceof Varien_Object) {
```

Comment:
Need to update docblock at line 306 `* @param   array || Varien_Object $column`.